cloudcredential/v1/types_aws.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2018 The OpenShift Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: these types should eventually be broken out, along with the actuator, to a separate repo.
+
+// AWSProviderSpec contains the required information to create a user policy in AWS.
+// +k8s:deepcopy-gen=false
```

Comment:
why?

Reply:
The field in AWSProviderSpec.StatementEntries[].PolicyCondition being of type `map[string]map[string]interface{}`. This means we cannot generate DeepCopy. The `interface{}` unfortunately snuck into our API while DeepCopy generation was mistakenly turned off (restored in https://github.com/openshift/cloud-credential-operator/pull/204 ).

Reply:
Just create custom deepcopy funcs for IAMPolicyCondition or even IAMPolicyConditionKeyValue. Doesn't that work? No need to disable generation for these normal structs.